pulsar-client/src/main/java/org/apache/pulsar/client/api/ConsumerBuilder.java

```diff
@@ -89,12 +89,19 @@
     /**
      * Specify a list of topics that this consumer will subscribe on.
      * <p>
-     * This argument is required when constructing the consumer.
      *
      * @param topicNames
      */
     ConsumerBuilder topics(List<String> topicNames);
 
+    /**
+     * Specify a pattern for topics that this consumer will subscribe on.
+     * <p>
+     *
+     * @param topicsPattern
+     */
+    ConsumerBuilder topicsPattern(Pattern topicsPattern);
```

Comment:
I would prefer to just take `String` here for the pattern, since most of the time the regex will be very simple. 

We can also offer both versions: 
```java
ConsumerBuilder topicsPattern(Pattern topicsPattern);
ConsumerBuilder topicsPattern(String topicsPattern);
```

Reply:
Thanks, will add it.

Reply:
@zhaijack Actually, I was thinking that probably, using a "glob" pattern should be enough in most cases. 

I would expect that almost all usages of this API would be like 
"topic-category-*" or very similar. 

I cannot really see a case in which a full blown regex is indeed needed.

What do you think?

Reply:
:) @merlimat , yes, seems most of the case is like that you mentioned. Not sure if user's use case would be different as we thought, It maybe better to provide a wider compatibility for api, seems Kafka also use this ` java.util.regex.Pattern`. 
How about leave it as you suggested above?:
```
ConsumerBuilder topicsPattern(Pattern topicsPattern);
ConsumerBuilder topicsPattern(String topicsPattern);
```

Reply:
+1 for having both `String` and `Pattern` builders.